icu4c/source/common/rbbidata.h

```diff
@@ -93,33 +94,40 @@ struct RBBIDataHeader {
 };
 
 
-
-struct  RBBIStateTableRow {
-    int16_t          fAccepting;    /*  Non-zero if this row is for an accepting state.   */
+template <typename T>
+struct RBBIStateTableRowT {
+    T                fAccepting;    /*  Non-zero if this row is for an accepting state.   */
                                     /*  Value 0: not an accepting state.                  */
                                     /*       -1: Unconditional Accepting state.           */
                                     /*    positive:  Look-ahead match has completed.      */
                                     /*           Actual boundary position happened earlier */
                                     /*           Value here == fLookAhead in earlier      */
                                     /*              state, at actual boundary pos.        */
-    int16_t          fLookAhead;    /*  Non-zero if this row is for a state that          */
+    T                fLookAhead;    /*  Non-zero if this row is for a state that          */
                                     /*    corresponds to a '/' in the rule source.        */
                                     /*    Value is the same as the fAccepting             */
                                     /*      value for the rule (which will appear         */
                                     /*      in a different state.                         */
-    int16_t          fTagIdx;       /*  Non-zero if this row covers a {tagged} position   */
+    T                fTagIdx;       /*  Non-zero if this row covers a {tagged} position   */
                                     /*     from a rule.  Value is the index in the        */
                                     /*     StatusTable of the set of matching             */
                                     /*     tags (rule status values)                      */
-    int16_t          fReserved;
-    uint16_t         fNextState[1]; /*  Next State, indexed by char category.             */
+    T                fReserved;
+    T                fNextState[1]; /*  Next State, indexed by char category.             */
                                     /*    Variable-length array declared with length 1    */
                                     /*    to disable bounds checkers.                     */
                                     /*    Array Size is actually fData->fHeader->fCatCount*/
                                     /*    CAUTION:  see RBBITableBuilder::getTableSize()  */
                                     /*              before changing anything here.        */
 };
 
+typedef RBBIStateTableRowT<int8_t> RBBIStateTableRowS8;
```

Comment:
Note: If 'S' stands for "signed" then that might be misleading if you have both signed and unsigned types. (see previous comment)